raft/raft_api.nim

```diff
@@ -9,62 +9,160 @@
 
 import types
 import protocol
-
-export types, protocol
-
-# Raft Node Public API procedures / functions
-proc RaftNodeCreateNew*[LogEntryDataType, SmStateType](                     # Create New Raft Node
-                  id: RaftNodeId, peers: RaftNodePeers,
-                  persistentStorage: RaftNodePersistentStorage,
-                  msgSendCallback: RaftMessageSendCallback): RaftNode[LogEntryDataType, SmStateType] =
-  discard
-
-proc RaftNodeLoad*[LogEntryDataType, SmStateType](
+import consensus_module
+import log_ops
+import ../db/kvstore_mdbx
```

Comment:
I think the persistence module should be a bit more optional. For example, the base library can define some kind of callback that's responsible for implementing the persistence and then the user can import a helper module implementing the callback with a particular database engine.